### PR TITLE
e2e - tokenfactory bindings test

### LINKED ISSUE #377: Token factory testing
## Background
- Following the token factory module being added to JUNO, we need more testing for this 

## Suggested Design
-  Have a test contract and test its interaction with the token factory module

### PR DESCRIPTION
**Pull request by Reecepbcups**
Closes #377 
Merges into e2e  branch (not main). Does not require approvals

- Ensures the wasm bindings works with a real contract, tested with e2e testing
- tested locally with `make test-e2e`
- e2e passes now in this branch

### CODE REVIEW

**Comment by vuong177** [comment]
hey @Reecepbcups  test-e2e still fail.

**Comment by Reecepbcups** [comment]
@vuong177 -> https://github.com/CosmosContracts/juno/pull/363#issuecomment-1319618904
Since I built off e2e this is why it fails. Will be fixed when this merges -> e2e-test -> main

**Comment by vuong177** [comment]
@Reecepbcups this issue https://github.com/CosmosContracts/juno/issues/371 for automate build docker image when we release  new version. I think e2e test should be passed in this PR.

**Comment by vuong177** [comment]
@Reecepbcups I also need to review tf contract code. Did you create a repo for it.

**Comment by Reecepbcups** [comment]
> @Reecepbcups I also need to review tf contract code. Did you create a repo for it ?

@vuong177 https://github.com/CosmWasm/token-bindings/tree/main/contracts/tokenfactory
osmosis's example contract was compiled and used

**Comment by Reecepbcups** [comment]
I have not touched .github folder, but above `every commit / test` fails for 
`API error (404): pull access denied for juno, repository does not exist or may require 'docker login': denied: requested access to the resource is denied`

**Comment by vuong177** [comment]
Need to seperate normal test and e2e test when run CI. Add a go tag could be fix this.

File: tests/e2e/e2e_test.go [hunk 1]
```diff
@@ -1,15 +1,21 @@
-//go:build e2e
```
Comment (vuong177) [inline tests/e2e/e2e_test.go:1]: hey why you delete this ? Test fail because you remove this line

**Review by vuong177 — CHANGES_REQUESTED** [review]

File: tests/e2e/e2e_test.go (hunk 1 above)
Comment (Reecepbcups) [reply tests/e2e/e2e_test.go:1]: Whoops! This was from when I was trying to get it to run locally. Reverting back to how it was before sorry!

File: tests/e2e/e2e_setup_test.go [hunk 2]
```diff
@@ -100,7 +99,10 @@ func (s *IntegrationTestSuite) SetupSuite() {
 
 	if str := os.Getenv(upgradeVersionEnv); len(str) > 0 {
 		upgradeSettings.Version = str
-		s.T().Log(fmt.Sprintf("upgrade version set to %s", upgradeSettings.Version))
+		s.T().Logf("upgrade version set to %s", upgradeSettings.Version)
+	} else {
+		upgradeSettings.Version = "v12"
```
Comment (vuong177) [inline tests/e2e/e2e_setup_test.go:104]: `upgradeSettings.Version` shouldn't be hardcoded

**Comment by vuong177** [comment]
@Reecepbcups hmm I don't see the upgrade test in e2e log.

**Comment by Reecepbcups** [comment]
> @Reecepbcups hmm I don't see the upgrade test in e2e log.

@vuong177 The v12 upgrade, deposit, and vote test? Done here -> https://github.com/CosmosContracts/juno/actions/runs/3558650155/jobs/5977470658#step:9:563
upgrades to height 70 then the token binding test start & all are successful at the end

Since its json, success check is `"\"code\":0,"` now instead of `code: 0` given all logs were that way

**Comment by vuong177** [comment]
LGTM

**Pull request merged**